CFE-371:[WIP] [Draft] Docs work for ALB Operator (#43744)
openshift/openshift-docs: networking/aws_load_balancer_operator/creating-multiple-ingress-through-single-alb.adoc

// Module included in the following assemblies:
//
// * networking/aws_load_balancer_operator/multiple-ingress-through-single-alb.adoc

:_content-type: PROCEDURE
[id="nw-creating-multiple-ingress-through-single-alb_{context}"]
= creating-multiple-ingress-through-single-alb

> The user can route the traffic to services as a part of a domain and can serve all the endpoints through a single ALB.

Reviewer: ```suggestion
The user can route the traffic to multiple Ingresses through a single ALB.
```

Author: Addressed

spec:
  ingressClass: alb <5>
  rules:
  - host: example.com <6>
    http:
        paths:
        - path: /blog <7>
          backend:
            service:
              name: example-1 <8>
              port:
                number: 80 <9>
  —--
kind: Ingress
metadata:
  name: example-2
  annotations:
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/group.name: example
    alb.ingress.kubernetes.io/group.order: "2"
spec:
  ingressClass: alb
  rules:
  - host: example.com
    http:
        paths:
        - path: /store
          backend:
            service:
              name: example-2
              port:
                number: 80
  —--
kind: Ingress
  metadata:
  name: example-3
  annotations:
    alb.ingress.kubernetes.io/scheme: internet-facing
    alb.ingress.kubernetes.io/group.name: example
    alb.ingress.kubernetes.io/group.order: "3"
spec:
  ingressClass: alb
  rules:
  - host: example.com
    http:
        paths:
        - path: /
          backend:
            service:
              name: example-3
              port:
                number: 80
----

<1> Defines name of an `Ingress`.

> <2>

Reviewer: This annotation indicates that the load balancer should be provisioned in the public subnet and makes it accessible over the internnet.

Author: Addressed

> <3>

Reviewer: The name of the _IngressGroup_ which ensures that all _Ingresses_ which belong the group are serviced by the same load balancer.

Author: Addressed

> <4>

Reviewer: This specifies the order in which the rules from the Ingresses are matched when the request is received at the load balancer.

Author: Addressed

> <5> Defines name of an `ingressClass`.

Reviewer: Specifies which `IngressClass` this _Ingress_ belongs to.

Author: Addressed

> <6> Defines name of a `domain`.

Reviewer: Specifies the domain which is used for request routing.

Author: Acked

> <7>

Reviewer: The path which should be routed to the service.

Author: Addressed

<8>

> <9>

Reviewer: The port on the service which service the endpoint.

Author: Addressed

> <9>

Reviewer: The name of the service which serves the endpoint configured in the Ingress.

Author: Addressed
